src/test/kotlin/com/gamingbacklog/api/gamingbacklogapi/unit/services/GameInstanceServiceTests.kt

```diff
@@ -0,0 +1,131 @@
+      Assertions.assertEquals(foundGame.platforms, result.platforms)
+      Assertions.assertEquals(foundGame.genres, result.genres)
+      Assertions.assertEquals(foundGame.universes, result.universes)
+      Assertions.assertEquals(foundGame.companies, result.companies)
+      Assertions.assertEquals(foundGame.releaseDate, result.releaseDate)
+      Assertions.assertEquals(foundGame.images, result.images)
+    }
+  }
+
+  @Nested
+  @DisplayName("Update With Custom Fields Tests")
+  inner class UpdateWithCustomFields {
+
+    // update some fields
+  @Test
+  fun shouldUpdateSomeFields() {
+      val gameInstanceRequest = GameInstanceRequest(
+        null,
+        9,
+        null,
+        null,
+        null,
+        null,
+        "wow space yay so fun",
+        arrayListOf("PS3")
+      )
+      val game = GameInstance(
+        "id", "igdb1", "Mass Effect 2",
+        arrayListOf("a lot"),
+        arrayListOf("RPG"), arrayListOf("Mass Effect"), arrayListOf("BioWare"),
+        arrayListOf("2007"), arrayListOf("normandy")
+      )
+      given(gameInstanceRepository.findOneById(any())).willReturn(game)
+      gameInstanceService.updateWithCustomFields("dd7f03b962f1f3416d08ee0f", gameInstanceRequest)
+      Assertions.assertEquals(game.rating, gameInstanceRequest.rating)
+      Assertions.assertEquals(game.review, gameInstanceRequest.review)
+      Assertions.assertEquals(game.ranking, gameInstanceRequest.ranking)
+      Assertions.assertEquals(game.yearPlayed, gameInstanceRequest.yearPlayed)
+      Assertions.assertEquals(game.yearReceived, gameInstanceRequest.yearReceived)
+      Assertions.assertEquals(game.notes, gameInstanceRequest.notes)
+      Assertions.assertEquals(game.platformsOwnedOn, gameInstanceRequest.platformsOwnedOn)
+    }
+
+  @Test
+  fun shouldUpdateAllFields() {
+    val gameInstanceRequest = GameInstanceRequest(
+      null,
+      8,
+      "nostalgia is cool but idk if this game is good anymore, oh well love it anyway",
+      "30",
+      2005,
+      2005,
+      "nostalgia games, need to replay before disney trip",
+      arrayListOf("PS2", "PS3", "PS4")
+    )
+    val game = GameInstance("id", "igdb1", "Kingdom Hearts",
+      arrayListOf("a lot"),
+      arrayListOf("RPG"), arrayListOf("Kingdom Hearts"), arrayListOf("Square Enix"),
+      arrayListOf("2002"), arrayListOf("heart in the light or something"))
+    given(gameInstanceRepository.findOneById(any())).willReturn(game)
+    gameInstanceService.updateWithCustomFields("dd7f03b962f1f3416d08ee0f", gameInstanceRequest)
+    Assertions.assertEquals(game.rating, gameInstanceRequest.rating)
+    Assertions.assertEquals(game.review, gameInstanceRequest.review)
+    Assertions.assertEquals(game.ranking, gameInstanceRequest.ranking)
+    Assertions.assertEquals(game.yearPlayed, gameInstanceRequest.yearPlayed)
+    Assertions.assertEquals(game.yearReceived, gameInstanceRequest.yearReceived)
+    Assertions.assertEquals(game.notes, gameInstanceRequest.notes)
```

Comment:
This is something to decide later probably but when a user adds notes to a game, will we be giving them the option to add them in addition to existing notes? Or will they be replacing what exists?

Reply:
Good question! I'm guessing we should let them add in addition to, but I think that will be later on/out of scope for this ticket.